### PR TITLE
feat: add generic OIDC authentication (Okta, Auth0, Keycloak, OneLogin, ...)

### PR DESCRIPTION
**Pull request by d-axel-b**
## Summary

Adds SSO login via any OIDC-compliant identity provider using better-auth's `genericOAuth` plugin. One integration that works with Okta, Auth0, Keycloak, OneLogin, and any provider that exposes a standard OIDC discovery document.

## Configuration

Set the following environment variables to enable:

- `OIDC_DISCOVERY_URL` (required) — provider's `.well-known/openid-configuration` URL
- `OIDC_CLIENT_ID` (required)
- `OIDC_CLIENT_SECRET` (required)
- `OIDC_PROVIDER_ID` — defaults to `oidc`, used in callback URL and button logic
- `OIDC_PROVIDER_NAME` — defaults to `SSO`, shown on the login button
- `OIDC_SCOPES` — defaults to `openid,profile,email`
- `OIDC_AUTH_DOMAINS` — optional comma-separated email domain allowlist
- `OIDC_PKCE` — defaults to `true`

When unset, the SSO button is hidden.

Redirect URI to register in your IdP: `https://<your-nao-host>/api/auth/oauth2/callback/{OIDC_PROVIDER_ID}`

## What's included

- **Backend:** `genericOAuth` plugin registration, env schema (8 vars), domain allowlist enforcement in `databaseHooks.user.create.before`, OIDC users included in `isSocial` check for auto-provisioning, tRPC `authConfig.oidc.getConfig` endpoint
- **Frontend:** `genericOAuthClient` plugin, dynamic `handleOidcSignIn` via `signIn.oauth2`, conditional "Continue with {providerName}" button with `LockKeyholeIcon`
- **Docs:** setup guide at `apps/backend/docs/auth-oidc.md` with walkthroughs for Okta, Auth0, Keycloak, and OneLogin
- **Tests:** 15 Vitest unit tests — 10 for domain allowlist edge cases, 5 for tRPC endpoint

## Test plan

- [x] `npm run lint` passes (no new errors — pre-existing drizzle-orm type issues unrelated)
- [x] `npm test` passes (15/15 new tests pass)
- [x] Manual smoke against an OIDC provider:
  - [x] New user → user + account rows created, session set, redirected to `/`
  - [x] Returning user → no new user row, existing session restored
  - [x] SSO button visible when env vars set, hidden when unset
  - [x] Custom provider name displayed on button
  
  Some manual test remains

### CODE REVIEW

**Comment by radcheb** [comment]
Related: #591 and #657

**Comment by wgenchi-mwb** [comment]
@Bl3f do you have some timeline when this will be available? okta integration is a critical feature for adoption in our case.. talked also with @ClaireGz about this during a call

**Comment by Bl3f** [comment]
hey @wgenchi-mwb this feature will be available through Enterprise version mid of next week!

**Comment by wgenchi-mwb** [comment]
@Bl3f what about an Okta integration for the self-hosted version? Currently only Google auth is available

**Comment by Bl3f** [comment]
@wgenchi-mwb it will be in the self-hosted but will require to pay for the Enterprise license (and set through the `NAO_LICENSE` env variable) — main reason is that we need to be paid for the product we develop and gatekeeping these features for companies requiring Okta, Azure Entra ID or row-level security seems fair as it's most of the time an Enterprise requirement. Happy to chat about it!

**Comment by d-axel-b** [comment]
✅ Tested and approved using my personnal okta developper/integrator account. 

I added the full procedure to run the tests in **apps/backend/docs/auth-oidc-local-testing.md**

Worth automation through CI/CD with a potential dedicated okta integrator account for nao.

-----

This first implementation does not handle groups management. But could be an evolution later.

**Comment by d-axel-b** [comment]
Tested with : Password / Auth Code / FastPass

------

SCR-20260522-mfgh

-----

SCR-20260522-ptpp

-----

SCR-20260522-pxrv

-----

SCR-20260522-pxbe

-----

SCR-20260522-pwsr

-----

SCR-20260522-pvdd

File: apps/backend/docs/auth-oidc.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# OIDC / SSO Authentication
```
Comment (Bl3f) [inline apps/backend/docs/auth-oidc.md:1]: We need to find a way to be able to import community maintained docs like this one to our internal private nao-docs repo that builds over at https://docs.getnao.io.

I'm not fond of having a specific doc here to be honest, but maybe we can firstly merge like this and we will move it in another PR.

File: apps/backend/src/services/oidc-auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import type { BetterAuthPlugin } from 'better-auth';
```
Comment (Bl3f) [inline apps/backend/src/services/oidc-auth.service.ts:3]: Would be great if you can add the license header for this file pls.

```
/* @license Enterprise */
```

File: apps/frontend/src/components/auth-form.tsx [hunk 3]
```diff
@@ -83,6 +86,17 @@ export function AuthForm({
...
 						)}
 						{isMicrosoftSetup && <MicrosoftSignInButton callbackUrl={socialCallbackUrl} />}
+						{oidcConfig.data && (
+							<Button
+								type='button'
+								variant='outline'
+								className='w-full h-11'
+								onClick={() => handleOidcSignIn(oidcConfig.data!.providerId, socialCallbackUrl)}
+							>
+								<LockKeyholeIcon className='w-5 h-5' />
```
Comment (Bl3f) [inline apps/frontend/src/components/auth-form.tsx]: We can eventually hardcode if the providerId is Okta the okta logo and defaults on LockKeyhole if not?

**Review by Bl3f — APPROVED** [review]
LGTM, thank you so much @d-axel-b added a few comments and will merge this over the weekend. This is awesome

**Comment by d-axel-b** [comment]
Added this behaviour for Okta, Auth0 and Keycloack, could add more oidc providers later: 

image

**Comment by d-axel-b** [comment]
@Bl3f - Please wait a little bit before merging.

I noticed unexpected behaviors about allowed domains and about user org attachment.

Looking at it.

[Fixed]

**Comment by d-axel-b** [comment]
Added a bug fix: OIDC users not auto-provisioned into org/project on signup

The databaseHooks.user.create hooks in auth.ts relied on ctx.params.id to identify the auth provider. This works for built-in social providers (Google, GitHub, Microsoft) which route through /api/auth/callback/:id, but the genericOAuth plugin (OIDC) routes through /api/auth/oauth2/callback/:providerId — exposing the provider ID as ctx.params.providerId instead.

This caused two issues for OIDC users:

- before hook: The OIDC_AUTH_DOMAINS email domain allowlist was silently bypassed — any domain could signin
- after hook: isSocial evaluated to false, so addUserToDefaultProjectIfExists was skipped — OIDC users were created in the user table but never added to the default org/project, making them invisible in the admin settings page

Fix: Extracted a resolveProviderId(ctx) utility that checks both ctx.params.id and ctx.params.providerId, used in both hooks.

**Comment by d-axel-b** [comment]
@Bl3f - Fix has been committed and pushed - everything seems ok on my side.

**Review by Bl3f — APPROVED** [review]
Thank you so much. I just tested it and it works like a charm, sorry for being a bit late to merge!
